governator-core/src/main/java/com/netflix/governator/LifecycleInjectorCreator.java

```diff
@@ -0,0 +1,152 @@
+package com.netflix.governator;
+
+import java.util.Arrays;
+import java.util.IdentityHashMap;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.ConfigurationException;
+import com.google.inject.CreationException;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.ProvisionException;
+import com.google.inject.Stage;
+import com.google.inject.TypeLiteral;
+import com.netflix.governator.annotations.SuppressLifecycleUninitialized;
+import com.netflix.governator.annotations.binding.Arguments;
+import com.netflix.governator.annotations.binding.Profiles;
+import com.netflix.governator.internal.DefaultPropertySource;
+import com.netflix.governator.internal.GovernatorFeatureSet;
+import com.netflix.governator.spi.InjectorCreator;
+import com.netflix.governator.spi.PropertySource;
+
+/**
+ * Custom strategy for creating a Guice Injector that enables support for lifecycle annotations such 
+ * as {@link @PreDestroy} and {@link @PostConstruct} as well as injector lifecycle hooks via the 
+ * {@link LifecycleListener} API. 
+ * 
+ * The LifecycleInjectorCreator may be overridden to handle pre-create and post-create notification.
+ */
+public class LifecycleInjectorCreator implements InjectorCreator<LifecycleInjector>, LifecycleListener {
```

Comment:
It may make more sense to rename governator-annotations to governator-api, and also move interfaces such as LifecycleListener there as well as the annotations.